### PR TITLE
refactor: react suspense log added

### PR DESCRIPTION
**Pull request by PritishBudhiraja**

### CODE REVIEW

File: src/orca-log-catcher/ErrorBoundary.res [hunk 1]
```diff
@@ -157,6 +157,7 @@ module ErrorCard = {
           firstEvent: false,
           metadata: JSON.Encode.null,
           ephemeralKey: "",
+          componentName,
```
Comment (ArushKapoorJuspay) [inline src/orca-log-catcher/ErrorBoundary.res:160]: @Sanskar2001 will we be able to see this new type directly in Grafana or some other changes will be required? The use of this type is to help detect the component which crashed

File: src/PaymentElement.res [hunk 2]
```diff
@@ -284,47 +284,47 @@ let make = (~cardProps, ~expiryProps, ~cvcProps, ~paymentType: CardThemeType.mod
     <PaymentShimmer />
   }
   let checkoutEle = {
-    <ErrorBoundary key={selectedOption}>
+    <ErrorBoundary key={selectedOption} componentName="PaymentElement">
```
Comment (ArushKapoorJuspay) [inline src/PaymentElement.res:287]: Do we need this ErrorBoundary?

File: src/orca-log-catcher/ReusableReactSuspense.res [hunk 3]
```diff
@@ -0,0 +1,8 @@
+@react.component
+let make = (~children, ~loaderComponent, ~componentName) => {
+  Console.log2("-- componentName -- ", componentName)
```
Comment (ArushKapoorJuspay) [inline src/orca-log-catcher/ReusableReactSuspense.res:3]: Please remove this console

File: src/orca-log-catcher/ErrorBoundary.res (hunk 1 above)
Comment (Sanskar2001) [reply src/orca-log-catcher/ErrorBoundary.res:160]: @PritishBudhiraja We don't want to introduce one more column in sdk_events for this, can you please append the component name in value only?

Comment (PritishBudhiraja) [reply src/orca-log-catcher/ErrorBoundary.res:160]: updated please check once

**Review by seekshiva — APPROVED** [review]

**Pull request merged**